pyproject.toml

```diff
@@ -1,7 +1,29 @@
+[tool.mypy]
+pretty = true
+ignore_missing_imports = false
+check_untyped_defs = true
+no_implicit_reexport = false
+disallow_untyped_defs = false # remove once the project is typed
+strict = true
+plugins = ["pydantic.mypy"]
+
+[[tool.mypy.overrides]]
+module = ["pandas", "sklearn.*", "joblib", "scipy.*", "mariadb"]
+ignore_missing_imports = true
+
+[tool.pyright]
+# used in Pylance extension for language server
+# type check is done by mypy, disable to avoid unwanted errors
+typeCheckingMode = "off"
+
 [tool.isort]
 profile = "black"
 
+[tool.ruff]
+line-length = 100
```

Comment:
my suggestion is to extend the line length as modern dev setups usually don't have a low display limit.
Also the code is less dense with less line breaks (even though the code in each line should be short anyways.

Reply:
+1